cmd/jujud/agent/machine.go

```diff
@@ -458,6 +469,11 @@ func (a *MachineAgent) Run(*cmd.Context) (err error) {
 		return errors.Annotate(err, "error upgrading server certificate")
 	}
 
+	// moved from NewMachineAgent here because the agent config could not be ready yet there.
```

Comment:
so long as we are sure this can only ever be called once, since it will panic if the various Register() functions are called a second time

Reply:
`registerPrometheusCollectors` was moved from cmd constructor here.
The `machine` command is run by `systemd` in IAAS or docker in k8s.
Should it be ok?